src/messaging/ReliableMessageManager.cpp

```diff
@@ -404,57 +449,29 @@ CHIP_ERROR ReliableMessageManager::SendFromRetransTable(RetransTableEntry * entr
     CHIP_ERROR err              = CHIP_NO_ERROR;
     ReliableMessageContext * rc = entry->rc;
 
-    // To trigger a call to OnSendError, set the number of transmissions so
-    // that the next call to ExecuteActions will abort this entry,
-    // restart the timer immediately, and ExitNow.
-
-    CHIP_FAULT_INJECT(FaultInjection::kFault_RMPSendError, entry->sendCount = static_cast<uint8_t>(rc->mConfig.mMaxRetrans + 1);
-                      entry->nextRetransTimeTick = 0; StartTimer(); ExitNow());
```

Comment:
We need to cleanup the RC from retransTable in case of error, we can't just return an error immediately. 

We also need to take care of exit conditionally to pass CI since no one else use this.

Reply:
OK.  Maybe file a followup issue to look into adding this back?

Reply:
This faultIntjection is not valid with current logic, it will throw an error without cleaning up the RC from the table. The original purpose of this fault injection is to test CRMP resent without a peer during unit testing. Now I use loopback interface to test CRMP resend.